### PR TITLE
🚨Source The Guardian API: Migrate to poetry

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Migrating to poetry

## Can this PR be safely reverted and rolled back?
- [x] YES 💚

### CODE REVIEW

File: airbyte-integrations/connectors/source-the-guardian-api/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+readme = "README.md"
+documentation = "https://docs.airbyte.com/integrations/sources/the-guardian-api"
+homepage = "https://airbyte.com"
+repository = "https://github.com/airbytehq/airbyte"
+[[tool.poetry.packages]]
+include = "source_the_guardian_api"
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
+airbyte-cdk = "0.80.0"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-the-guardian-api/pyproject.toml:20]:
```suggestion
airbyte-cdk = "0.7.1"
```

**Comment by natikgadzhi** [comment]
`spec` fails — we should fix this up

**Comment by ChristoGrab** [comment]
Looks like the version of pyaml that comes bundled with the pinned CDK version (0.7.1) is failing to install and causing the whole build to fail

**Comment by ChristoGrab** [comment]
Had to thread the needle a bit with CDK versions to get one that worked, the connector is now pinned to CDK version "0.60.0". There are some CAT test failures but they are complaining about missing Record Count and legacy state, which should not be an issue for this version of the CDK.

**Review by ChristoGrab — APPROVED** [review]

**Pull request merged**